packages/design-carousel/src/index.tsx

```diff
@@ -1,45 +1,121 @@
-import { useStarterDesignsQuery } from '@automattic/data-stores';
+import { Gridicon } from '@automattic/components';
+import { StarterDesigns, useStarterDesignsQuery } from '@automattic/data-stores';
 import { useLocale } from '@automattic/i18n-utils';
 import { Button } from '@wordpress/components';
+import { useMediaQuery } from '@wordpress/compose';
 import { Icon, chevronLeft, chevronRight } from '@wordpress/icons';
 import { useI18n } from '@wordpress/react-i18n';
-import React, { useEffect, useRef } from 'react';
+import { useEffect, useRef } from 'react';
 import Swiper from 'swiper';
 import { Item } from './item';
 import 'swiper/dist/css/swiper.css';
 import type { Design } from '@automattic/design-picker/src/types';
+import type { MShotsOptions } from '@automattic/onboarding';
 
 type DesignCarouselProps = {
 	onPick: ( design: Design ) => void;
 	selectedDesignSlugs?: string[];
+	flow?: string;
 };
 
-export default function DesignCarousel( { onPick, selectedDesignSlugs }: DesignCarouselProps ) {
+export default function DesignCarousel( {
+	onPick,
+	selectedDesignSlugs,
+	flow,
+}: DesignCarouselProps ) {
 	const { __ } = useI18n();
-
 	const swiperInstance = useRef< Swiper | null >( null );
-
 	const locale = useLocale();
+	const isLargerThan1440px = useMediaQuery( '(min-width: 1440px)' );
+	const mobileOptions: MShotsOptions = { w: 400, vpw: 400, vph: 872, format: 'png' };
+	const desktopOptions: MShotsOptions = {
+		w: isLargerThan1440px ? 1920 : 1280,
+		vpw: 1920,
+		vph: 1280,
+		format: 'png',
+	};
 
 	const { data: allDesigns } = useStarterDesignsQuery( {
 		_locale: locale,
 	} );
 
-	let selectedDesigns = allDesigns?.designs;
+	const shouldDisplayMobileViewOnly = ( flow: string | undefined ) => {
+		switch ( flow ) {
+			case 'link-in-bio':
```

Comment:
Let's use [`isLinkInBioFlow`](https://github.com/Automattic/wp-calypso/blob/trunk/packages/onboarding/src/utils/flows.ts#LL28C14-L28C29) instead. And since this function doesn't depend on the state, we can hoist it up and outside the component; to avoid recreating it at every render.